staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -245,6 +283,34 @@ func (r *withRetry) After(ctx context.Context, request *Request, resp *http.Resp
 	}
 }
 
+func (r *withRetry) WrapPreviousError(finalErr error) error {
+	if finalErr == nil || r.previousErr == nil {
+		return finalErr
+	}
+	if !isContextError(finalErr) {
+		return finalErr
+	}
+
+	previousErr := r.previousErr
+	if err := errors.Unwrap(r.previousErr); err != nil {
```

Comment:
Why do we unwrap the previous before wrapping?  Is there some spot that always wraps and we need to remove a layer?  I expected to have the unwrapping happen by the caller (if at all).

Reply:
net/http wraps the underlying error with an url.Error object
https://github.com/golang/go/blob/29866aa2b6a6a48672a386d05b53779e768c13d6/src/net/http/client.go#L632-L636

I changed the bits here to unwrap only when the previous `err` is an instance of `url.Error` so we avoid error message like:
```
Error: Get "http://foo.bar/api/v1": context deadline exceeded - error from a previous attempt: Error: Get "http://foo.bar/api/v1": EOF
```
the error message will contain the url/op twice if we don't unwrap the `url.Error` from the previous `err`